packages/frontend/src/features/metricFlow/utils/convertMetricFlowFieldsToExplore.ts

```diff
@@ -18,30 +18,53 @@ export default function convertMetricFlowFieldsToExplore(
     tableName: string,
     metricFlowFields: GetMetricFlowFieldsResponse,
 ) {
-    const dimensionsMap = metricFlowFields.dimensions.reduce(
-        (acc, { name, description, type }) => {
-            const isTimeDimension = type === MetricFlowDimensionType.TIME;
-            const dimension: CompiledDimension = {
-                fieldType: FieldType.DIMENSION,
-                type: isTimeDimension
-                    ? DimensionType.TIMESTAMP
-                    : DimensionType.STRING,
-                // Note: time columns in results are suffixed with '__day' by default
-                name: isTimeDimension ? `${name}__day` : name,
-                description,
-                label: friendlyName(name),
-                table: tableName,
-                tableLabel: '',
-                sql: '',
-                compiledSql: '',
-                tablesReferences: [tableName],
-                hidden: false,
-            };
+    const dimensionsMap = metricFlowFields.dimensions.reduce<
+        Record<string, CompiledDimension>
+    >((acc, { name, description, type, queryableGranularities }) => {
+        const isTimeDimension = type === MetricFlowDimensionType.TIME;
+        const dimension: CompiledDimension = {
+            fieldType: FieldType.DIMENSION,
+            type: isTimeDimension
+                ? DimensionType.TIMESTAMP
+                : DimensionType.STRING,
+            name: name,
+            description,
+            label: friendlyName(name),
+            table: tableName,
+            tableLabel: '',
+            sql: '',
+            compiledSql: '',
+            tablesReferences: [tableName],
+            hidden: false,
+        };
+        acc[fieldId(dimension)] = dimension;
 
-            return { ...acc, [fieldId(dimension)]: dimension };
-        },
-        {},
-    );
+        if (isTimeDimension && queryableGranularities) {
+            queryableGranularities.forEach((timeGranularity) => {
+                const timeDimension: CompiledDimension = {
+                    fieldType: FieldType.DIMENSION,
+                    type: DimensionType.TIMESTAMP,
+                    // Note: time columns in results are suffixed with the time granularity eg:'__day'
+                    name: `${name}__${timeGranularity
+                        .toString()
+                        .toLowerCase()}`,
+                    description,
+                    label: `${friendlyName(name)} (${friendlyName(
```

Comment:
Wait for the other PR to be merged https://github.com/lightdash/lightdash/pull/7384
So we can use the correct labels.